tracer/test/Datadog.Trace.Security.IntegrationTests/AspNetCore5AsmToggle.cs

```diff
@@ -25,40 +26,56 @@ namespace Datadog.Trace.Security.IntegrationTests
 {
     public class AspNetCore5AsmToggle : AspNetBase, IDisposable
     {
+        private const string LogFileNamePrefix = "dotnet-tracer-managed-";
+
         public AspNetCore5AsmToggle(ITestOutputHelper outputHelper)
             : base("AspNetCore5", outputHelper, "/shutdown", testName: nameof(AspNetCore5AsmToggle))
         {
             SetEnvironmentVariable(ConfigurationKeys.Rcm.PollInterval, "500");
         }
 
+        // TODO addjust third parameter as the following PRs are merged:
+        // * https://github.com/DataDog/dd-trace-dotnet/pull/3120
+        // * https://github.com/DataDog/dd-trace-dotnet/pull/3171
+        // the verify file names will need adjusting too
         [SkippableTheory]
-        [InlineData(true, ApplyStates.ACKNOWLEDGED)]
-        [InlineData(false, ApplyStates.UNACKNOWLEDGED)]
-        [InlineData(null, ApplyStates.ACKNOWLEDGED)]
+        [InlineData(true, ApplyStates.ACKNOWLEDGED,  RcmCapablitiesIndices.AsmActivation)] // RcmCapablitiesIndices.AsmActivation | RcmCapablitiesIndices.AsmIpBlocking | RcmCapablitiesIndices.AsmDdRules)]
+        [InlineData(false, ApplyStates.UNACKNOWLEDGED, 0)] // RcmCapablitiesIndices.AsmIpBlocking | RcmCapablitiesIndices.AsmDdRules)]
+        [InlineData(null, ApplyStates.ACKNOWLEDGED, RcmCapablitiesIndices.AsmActivation)] // RcmCapablitiesIndices.AsmActivation | RcmCapablitiesIndices.AsmIpBlocking | RcmCapablitiesIndices.AsmDdRules)]
         [Trait("RunOnWindows", "True")]
-        public async Task TestSecurityToggling(bool? enableSecurity, uint expectedState)
+        public async Task TestSecurityToggling(bool? enableSecurity, uint expectedState, byte expectedCapablities)
         {
             var url = "/Health/?[$slice]=value";
             var agent = await RunOnSelfHosted(enableSecurity);
-            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity, expectedState);
+            var settings = VerifyHelper.GetSpanVerifierSettings(enableSecurity, expectedState, expectedCapablities);
+
+            using var logEntryWatcher = new LogEntryWatcher($"{LogFileNamePrefix}{SampleProcessName}*");
 
             var spans1 = await SendRequestsAsync(agent, url);
 
             agent.SetupRcm(Output, new[] { ((object)new Features() { Asm = new Asm() { Enabled = false } }, "1") }, "FEATURES");
 
             var request1 = await agent.WaitRcmRequestAndReturnLast();
             CheckAckState(request1, expectedState, null, "First RCM call");
-            // even the request show the applied state seems extra time is needed before it's active
-            await Task.Delay(1500);
+            CheckCapabilities(request1, expectedCapablities, "First RCM call");
+            if (enableSecurity == true)
+            {
+                await logEntryWatcher.WaitForLogEntry("AppSec Disabled");
+                await Task.Delay(1500);
```

Comment:
I'm unsure why this is needed, but it seems to make the CI a lot more stable (it always passes locally).

My initial thought that it was dues to the async rejit request seems to be wrong as the rejitted methods are only needed for handling the body.